### PR TITLE
fixed token gen bugs

### PR DESCRIPTION
**Pull request by ruks**

### CODE REVIEW

File: components/auth/org.wso2.carbon.auth.user.store/pom.xml [hunk 1]
```diff
@@ -79,6 +79,7 @@
             javax.crypto.*,
             org.wso2.carbon.datasource.core.*;version="${datasources.core.package.import.version.range}",
             org.wso2.carbon.kernel.*,
+            org.wso2.carbon.auth.core.*,
```
Comment (harsha89) [inline components/auth/org.wso2.carbon.auth.user.store/pom.xml:82]: we need to have a version import range defined

File: components/auth/org.wso2.carbon.auth.user.mgt/src/main/java/org/wso2/carbon/auth/user/mgt/impl/JDBCUserStoreManager.java [hunk 2]
```diff
@@ -40,7 +43,18 @@
     UserStoreConnector userStoreConnector;
 
     public JDBCUserStoreManager() {
-        this.userStoreConnector = ComponentHolder.getUserStoreConnector();
+        UserStoreConnector connector = UserStoreConnectorFactory.getUserStoreConnector();
+        UserStoreConfiguration config = new UserStoreConfiguration();
+        Map<String, Object> properties = new HashMap<>();
+        properties.put(JDBCConnectorConstants.DATA_SOURCE, "WSO2UM_DB");
```
Comment (harsha89) [inline components/auth/org.wso2.carbon.auth.user.mgt/src/main/java/org/wso2/carbon/auth/user/mgt/impl/JDBCUserStoreManager.java:49]: shall we used constants here

File: components/auth/org.wso2.carbon.auth.user.mgt/src/main/java/org/wso2/carbon/auth/user/mgt/impl/JDBCUserStoreManager.java [hunk 3]
```diff
@@ -40,7 +43,18 @@
...
+        properties.put(JDBCConnectorConstants.DATABASE_TYPE, "MySql");
```
Comment (harsha89) [inline components/auth/org.wso2.carbon.auth.user.mgt/src/main/java/org/wso2/carbon/auth/user/mgt/impl/JDBCUserStoreManager.java:50]: databsse type is hard coded?

**Review by malinthaprasan — APPROVED** [review]

File: components/auth/org.wso2.carbon.auth.user.mgt/src/main/java/org/wso2/carbon/auth/user/mgt/impl/JDBCUserStoreManager.java [hunk 4]
```diff
@@ -40,7 +43,18 @@
...
+        config.setProperties(properties);
+        try {
+            connector.init(config);
+        } catch (UserStoreConnectorException e) {
+            log.error("Error occurred while init UserStoreConnector", e);
```
Comment (harsha89) [inline components/auth/org.wso2.carbon.auth.user.mgt/src/main/java/org/wso2/carbon/auth/user/mgt/impl/JDBCUserStoreManager.java:55]: continue the flow even we encounter an error?

File: components/auth/org.wso2.carbon.auth.user.mgt/pom.xml [hunk 5]
```diff
@@ -37,5 +37,26 @@
...
+            org.eclipse.osgi.util,
+            org.slf4j.*;version="${slf4j.logging.package.import.version.range}",
+            org.osgi.framework.*;version="${osgi.framework.package.import.version.range}",
+            org.eclipse.osgi.framework.console;version="${osgi.framework.console.package.import.version.range}",
+            org.osgi.service.*;version="${equinox.osgi.services.package.import.version.range}",
+            org.osgi.util.tracker; version="${osgi.service.tracker.package.import.version.range}",
+            org.wso2.carbon.datasource.core.*;version="${datasources.core.package.import.version.range}",
+            com.zaxxer.hikari.*; version="${hikari.package.import.version.range}",
+            org.wso2.carbon.kernel.*;version="${carbon.kernel.package.import.version.range}",
+            <!--org.apache.axiom.*; version="${imp.pkg.version.axiom}"-->
```
Comment (harsha89) [inline components/auth/org.wso2.carbon.auth.user.mgt/pom.xml:52]: remove if this do not need

File: components/auth/org.wso2.carbon.auth.core/src/main/java/org/wso2/carbon/auth/core/encryption/SymmetricEncryption.java [hunk 6]
```diff
@@ -69,16 +69,19 @@ public void generateSymmetricKey() throws CryptoException {
 
         try {
 
-            ClassLoader classLoader = getClass().getClassLoader();
-            File file = new File(classLoader.getResource(CryptoConstants.SYMMETRIC_KEY_PROPERTIES_FILE_NAME).getFile());
+            String path = System.getProperty("wso2.runtime.path") + File.separator + "resources/auth/"
```
Comment (harsha89) [inline components/auth/org.wso2.carbon.auth.core/src/main/java/org/wso2/carbon/auth/core/encryption/SymmetricEncryption.java:72]: resources/auth/  has file separators

**Pull request merged**